storage/store.go

```diff
@@ -1198,97 +1198,100 @@ func (s *Store) Descriptor() (*proto.StoreDescriptor, error) {
 		reply.Reset()
 
 		// Get range and add command to the range for execution.
-		rng, err := s.GetRange(header.RaftID)
+		rng, err := s.GetRange(argsHeader.RaftID)
 		if err != nil {
-			reply.Header().SetGoError(err)
-			return retry.Break, err
-		}
-
-		if err = rng.AddCmd(ctx, proto.Call{Args: args, Reply: reply}, true); err == nil {
 			return retry.Break, err
 		}
 
-		// Maybe resolve a potential write intent error. We do this here
-		// because this is the code path with the requesting client
-		// waiting. We don't want every replica to attempt to resolve the
-		// intent independently, so we can't do it in Range.executeCmd.
-		if wiErr, ok := err.(*proto.WriteIntentError); ok {
-			pushType := proto.PUSH_TIMESTAMP
-			if proto.IsWrite(args) {
-				pushType = proto.ABORT_TXN
+		if err = rng.AddCmd(ctx, proto.Call{Args: args, Reply: reply}, true); err != nil {
+			// Maybe resolve a potential write intent error. We do this here
+			// because this is the code path with the requesting client
+			// waiting. We don't want every replica to attempt to resolve the
+			// intent independently, so we can't do it in Range.executeCmd.
+			if wiErr, ok := err.(*proto.WriteIntentError); ok {
+				var pushType proto.PushTxnType
+				if proto.IsWrite(args) {
+					pushType = proto.ABORT_TXN
+				} else {
+					pushType = proto.PUSH_TIMESTAMP
+				}
+				err = s.resolveWriteIntentError(ctx, wiErr, rng, args, pushType, false /* wait */)
 			}
-			err = s.resolveWriteIntentError(ctx, wiErr, rng, args, pushType, false /* wait */)
-			reply.Header().SetGoError(err)
-		}
 
-		switch t := err.(type) {
-		case *proto.WriteTooOldError:
-			// Update request timestamp and retry immediately.
-			header.Timestamp = t.ExistingTimestamp.Next()
-			return retry.Reset, err
-		case *proto.WriteIntentError:
-			// If write intent error is resolved, exit retry/backoff loop to
-			// immediately retry.
-			if t.Resolved {
+			switch t := err.(type) {
+			case *proto.WriteTooOldError:
+				// Update request timestamp and retry immediately.
+				argsHeader.Timestamp = t.ExistingTimestamp.Next()
 				return retry.Reset, err
-			}
+			case *proto.WriteIntentError:
+				// If write intent error is resolved, exit retry/backoff loop to
+				// immediately retry.
+				if t.Resolved {
+					return retry.Reset, err
+				}
 
-			// Otherwise, update timestamp on read/write and backoff / retry.
-			for _, intent := range t.Intents {
-				if proto.IsWrite(args) && header.Timestamp.Less(intent.Txn.Timestamp) {
-					header.Timestamp = intent.Txn.Timestamp.Next()
+				// Otherwise, update timestamp on read/write and backoff / retry.
+				for _, intent := range t.Intents {
+					if proto.IsWrite(args) && argsHeader.Timestamp.Less(intent.Txn.Timestamp) {
+						argsHeader.Timestamp = intent.Txn.Timestamp.Next()
+					}
 				}
+				return retry.Continue, err
 			}
-			return retry.Continue, err
 		}
-		return retry.Break, err
```

Comment:
It looks like you're missing a `return retry.Break, err` line. If err is not nil but it was not one of the types handled in the switch above, we still need to return it.